feat: add oracle linux

virtualistic opened the pull request:
# <!-- Intentionally left blank. -->

## Summary of Pull Request


As I requested here:
https://github.com/vmware-samples/packer-examples-for-vsphere/discussions/659 and inspired by your suggestion, created a new build for Oracle Linux 8.  Had to change the menu a bit to "make room" for the Oracle 8 template at position 19. (I skipped position 18 for Oracle 9 in the future). I was able to successfully build a new Oracle 8 template but hopefully you have some time to test and verify this.

## Type of Pull Request

<!--
    Please check the one that applies to this pull request using "[x]".
-->

- [ ] This is a bugfix. `type/bug`
- [X] This is an enhancement or feature. `type/feature` or `type/enhancement`
- [ ] This is a documentation update. `type/docs`
- [ ] This is a refactoring update. `type/refactor`
- [ ] This is a chore. `type/chore`
- [ ] This is something else.
      Please describe:

## Related to Existing Issues

<!--
  Is this related to any GitHub issue(s)?

  For example:
  - Resolves #1234
-->

Closes #671

## Test and Documentation Coverage

<!--
    Please check the one that applies to this pull request using "[x]".
-->

- [X] Tests have been completed.
- [ ] Documentation has been added or updated.

## Breaking Changes?

<!--
    Please check the one that applies to this pull request using "[x]".
-->

- [ ] Yes, there are breaking changes.
- [X] No, there are no breaking changes.

<!--
    If this pull request contains a breaking change, please describe the impact and mitigation path.
-->

tenthirtyam reviewed (commented):
The build.yaml needs to be updated. This is used to generated the build.sh with gomplate.

The README.md would also need an update.

tenthirtyam commented:
Marked as drafted until the CLA is approved.

virtualistic commented:
I added a new commit which added Oracle Linux as an entry to the README
Also added an entry for build.yaml

tenthirtyam reviewed (changes requested):
Hi @virtualistic -

Please run the following to generate the `build.sh`:

```bash
gomplate -c build.yaml -f build.tmpl -o build.sh
```

Additionally, in the `README.md` and update is needed to list the download information for each the distribution:

**Ryan Johnson**
Senior Staff Solutions Architect | Product Engineering @ VMware, Inc.

virtualistic commented:
Created a new commit that:
- Rebuilt the build.sh using the command:
`gomplate -c build.yaml -f build.tmpl -o build.sh`

- Added the source for the Oracle ISO to the README

tenthirtyam commented:
Thanks! Once the CLA is approved I'll run the standard tests. I have a couple updates and pull requests that I have in my backlog for this side project but will prioritize as soon as possible.

Might I ask you to open an enhancement issue and link the pull request with "Closes #x" when time permits.

virtualistic commented:
> Thanks! Once the CLA is approved I'll run the standard tests. I have a couple updates and pull requests that I have in my backlog for this side project but will prioritize as soon as possible.
> 
> Might I ask you to open an enhancement issue and link the pull request with "Closes #x" when time permits.

Sure, no problem.   ( Enhancement request has a typo btw: "Please add a short _decsription_." )
When  I open a new Enhancement Request, there are several **mandatory** fields:
Description
Use Case(s)
Potential Configuration
Do I type:  "closes  # 670" in each field or do you want that placed in a certain field?

tenthirtyam commented:
Closes is put here in the PR back to the enhancement issue.

thanks for the note on the issue form! :)

virtualistic commented:
closes #671

virtualistic commented:
> Closes is put here in the PR back to the enhancement issue.
> 
> thanks for the note on the issue form! :)

Most welcome!
Please note that the Oracle iso has an annoying mix of CaPs in its filename which I changed to all lowercase after I download the iso and also used the lowercase name in the packer file that references the iso.
So 2 questions:
1) Do you want me to use the original filename of the ISO in this pull request?

2) I also changed the iso_path to `iso/misc` for my situation in the `linux-oel.auto.pkrvars.hcl` . Do you want me to change it to `iso/linux/oracle` for this pull request?

tenthirtyam commented:
Please use the original file name from the vendor and an oel or oracle path.

virtualistic commented:
Added a new commit with the iso_path in line with the rest of the project and replaced the iso name with that of the vendor.

tenthirtyam commented:
Would you consider including OEL9, too?

tenthirtyam commented on README.md:
```diff
@@ -843,6 +844,8 @@ The project supports using a datastore to store your guest operating system [`.i
    - SUSE Linux Enterprise 15
      - [Download][download-linux-sles-15] the latest 15.4 release of the **FULL** `.iso` image. (_e.g.,_ `SLE-15-SP4-Full-x86_64-GM-Media1.iso`)
+   - Oracle Enterprise Linux 8
+     - [Download][download-linux-oel-8] the latest release of the **FULL** `.iso` image. (_e.g.,_ `oraclelinux-r8-u8-x86_64-dvd.iso`)
```
```suggestion
     - [Download][download-linux-oel-8] the latest release of the **FULL** `.iso` image. (_e.g.,_ `OracleLinux-R8-U8-x86_64-dvd.iso`)
```

tenthirtyam commented on builds/linux/oracle/8/variables.pkr.hcl:
```diff
@@ -0,0 +1,377 @@
+
+variable "vm_guest_os_name" {
+  type        = string
+  description = "The guest operating system name. Used for naming . (e.g. 'rhel')"
```
```suggestion
  description = "The guest operating system name. Used for naming . (e.g. 'oel')"
```

tenthirtyam commented on builds/linux/oracle/8/variables.pkr.hcl:
```diff
@@ -0,0 +1,377 @@
+
+variable "vm_guest_os_type" {
+  type        = string
+  description = "The guest operating system type, also know as guestid. (e.g. 'rhel8_64Guest')"
```
```suggestion
  description = "The guest operating system type, also know as guestid. (e.g. 'oracleLinux8_64Guest')"
```

tenthirtyam commented on builds/linux/oracle/8/variables.pkr.hcl:
```diff
@@ -0,0 +1,377 @@
+
+variable "iso_path" {
+  type        = string
+  description = "The path on the source vSphere datastore for ISO image. (e.g. 'iso/linux/rhel')"
```
```suggestion
  description = "The path on the source vSphere datastore for ISO image. (e.g. 'iso/linux/oel')"
```

tenthirtyam commented on builds/linux/oracle/8/variables.pkr.hcl:
```diff
@@ -0,0 +1,377 @@
+
+variable "iso_file" {
+  type        = string
+  description = "The file name of the ISO image used by the vendor. (e.g. 'rhel-<verssion>-x86_64-dvd.iso')"
```
```suggestion
  description = "The file name of the ISO image used by the vendor. (e.g. 'OracleLinux-<version>-x86_64-dvd.iso')"
```

tenthirtyam reviewed (changes requested):
Minor changes.

Would you consider adding OEL 9?

virtualistic commented:
Don't know if v9 will work because the latest vCenter v7 does **not**  offer Oracle Linux 9 as a guest os . 
(while my local ESXi server does)
![2023-09-21 21_29_16-vCenter-oracleinux-versions](https://github.com/vmware-samples/packer-examples-for-vsphere/assets/19362526/d1420389-0694-4d22-a26c-2ed8a5963564)

virtualistic commented:
I agree with your suggested changes. Think it's best I edit my repo and create a new commit with the changes? 
Or should I do a "signoff and commit suggestion" from the GitHub UI?

tenthirtyam commented:
> I agree with your suggested changes. Think it's best I edit my repo and create a new commit with the changes?
> Or should I do a "signoff and commit suggestion" from the GitHub UI?

 "signoff and commit suggestion" is fine, but I generally just amend the last commit and force push.

tenthirtyam commented:
> Don't know if v9 will work because the latest vCenter v7 does **not** offer Oracle Linux 9 as a guest os . (while my local v7 ESXi server does).

Use `other5xLinux64Guest`. 😃

virtualistic commented:
Ok, I changed all the requested files and did a :
```
git add -A
git commit --amend --no-edit
git push -f
```
Will have look if it's easy to add Oracle 9 now using my own Oracle 8 files :)

virtualistic commented:
Ok, pushed a new commit with an Oracle 9 Template.
![2023-09-22 13_56_08-oel9](https://github.com/vmware-samples/packer-examples-for-vsphere/assets/19362526/5f885f5b-3dab-48b3-8746-1917eaece6f4)

tenthirtyam commented:
Awesome! thank you!!

tenthirtyam reviewed (approved):
LGTM!

tenthirtyam reviewed (approved):
LGTM!
